src/archivematicaCommon/lib/elasticSearchFunctions.py

```diff
@@ -206,20 +219,13 @@ def aip_mapping_is_correct(conn):
 
     return mapping['aipfile']['properties']['AIPUUID']['index'] == 'not_analyzed'
 
-# try up to three times to get a connection
-def connect_and_create_index(index, attempt=1):
-    if attempt <= 3:
-        conn = Elasticsearch(hosts=getElasticsearchServerHostAndPort())
-
-        response = conn.indices.create(index, ignore=400)
-        if 'error' in response and 'IndexAlreadyExistsException' in response['error']:
-            return conn
-
-        set_up_mapping(conn, index)
-        conn = connect_and_create_index(index, attempt + 1)
-    else:
-        conn = False
-
+def connect_and_create_index(index):
+    conn = Elasticsearch(hosts=getElasticsearchServerHostAndPort())
+    conn.indices.create(index, ignore=400)
+    # Always put the mapping, even if the index already exists, because
+    # there may have been changes to the mapping structure; rerunning this
+    # will allow additional fields to be added to the index
```

Comment:
Heh, not sure.  I was looking at http://elasticsearch-py.readthedocs.org/en/master/api.html?highlight=ignore#ignore which suggested 404 might be useful, but possibly not in this context.